astroquery/gaia/core.py

```diff
@@ -549,5 +698,356 @@ def __getCoordInput(self, value, msg):
+                     table_description=None,
+                     format=None, verbose=False):
+        """Uploads a table to the user private space
+
+        Parameters
+        ----------
+        upload_resource : object, mandatory
+            resource to be uploaded (table, file or URL)
+        table_name: str, required if uploadResource is provided, default None
+            resource temporary table name associated to the uploaded resource
+        table_description: str, optional, default None
+            table description
+        format : str, optional, default 'votable'
+            results format
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+        """
+        return self.__gaiatap.upload_table(
+            upload_resource=upload_resource,
+            table_name=table_name,
+            table_description=table_description,
+            format=format, verbose=verbose)
+
+    def upload_table_from_job(self, job=None, table_name=None,
+                              table_description=None, verbose=False):
+        """Uploads a table to the user private space from a job
+
+        Parameters
+        ----------
+        job : job, mandatory
+            job used to create a table
+        table_name: str, required if uploadResource is provided, default None
+            resource temporary table name associated to the uploaded resource
+        table_description: str, optional, default None
+            table description
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+        """
+
+        return self.__gaiatap.upload_table_from_job(job,
+                                                    table_name,
+                                                    table_description,
+                                                    verbose)
+
+    def update_user_table(self, table_name=None, list_of_changes=[],
+                          verbose=False):
+        """Updates a user table
+
+        Parameters
+        ----------
+        table_name: str, required
+            table to be updated
+        list_of_changes: list, required
+            list of lists, each one of them containing sets of
+            [column_name, field_name, value].
+            column_name is the name of the column to be updated
+            field_name is the name of the tap field to be modified
+            field name can be 'utype', 'ucd', 'flags' or 'indexed'
+            value is the new value this field of this column will take
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+        """
+        return self.__gaiatap.update_user_table(table_name,
+                                                list_of_changes,
+                                                verbose)
+
+    def set_ra_dec_columns(self, table_name=None,
```

Comment:
@jcsegovia There are several functions just like this one which duplicate the functionality of `TapPlus`.  It looks like a more straightforward approach, which would avoid all of this duplicate code, would be to have `GaiaClass` inherit from `TapPlus`.  Is there a reason this is not done?  If you want to discuss this in more detail live (e.g., via slack), let me know, I can arrange a time.

Reply:
@keflavich you are right. I think it is a good idea to extend TapPlus, so we can remove duplicated code. Thanks!